view/src/main/java/cmm/apps/esmorga/view/eventdetails/EventDetailsViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package cmm.apps.esmorga.view.eventdetails
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import cmm.apps.esmorga.domain.event.GetEventDetailsUseCase
+import cmm.apps.esmorga.view.eventdetails.model.EventDetailsEffect
+import cmm.apps.esmorga.view.eventdetails.model.EventDetailsUiState
+import cmm.apps.esmorga.view.eventlist.mapper.EventUiMapper.toEventUiDetails
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class EventDetailsViewModel(app: Application, private val getEventDetailsUseCase: GetEventDetailsUseCase) : AndroidViewModel(app) {
+    private val _uiState = MutableStateFlow(EventDetailsUiState())
+    val uiState: StateFlow<EventDetailsUiState> = _uiState.asStateFlow()
+
+    private val _effect: MutableSharedFlow<EventDetailsEffect> = MutableSharedFlow(extraBufferCapacity = 2, onBufferOverflow = BufferOverflow.DROP_OLDEST)
+    val effect: SharedFlow<EventDetailsEffect> = _effect.asSharedFlow()
+
+    fun loadEventDetails(eventId: String) {
+        viewModelScope.launch {
+            val result = getEventDetailsUseCase(eventId)
+            result.onSuccess {
+                _uiState.value = it.data.toEventUiDetails()
+            }
+        }
+    }
+
+    fun onNavigateClick() {
+        _effect.tryEmit(
+            EventDetailsEffect.NavigateToLocation(
+                uiState.value.locationLat!!,
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/cmm-apps-android/EsmorgaAndroid/pull/5#issuecomment-2194728801>#267d9d</a><span> - 06/27/2024, 01:47 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The 'onNavigateClick' function uses non-null asserted calls ('!!') on 'locationLat' and 'locationLng'. If these values are null, it will cause a crash.
</div>
<div id="fix">
<b>Fix:</b> Ensure that 'locationLat' and 'locationLng' are not null before using them. If they are null, handle the case appropriately, possibly by not emitting the navigation effect.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -37,7 +37,11 @@
         EventDetailsEffect.NavigateToLocation(
             uiState.value.locationLat!!,
             uiState.value.locationLng!!
         )
     )
 +   } else {
 +       // Handle the case where locationLat or locationLng is null
 +       _effect.tryEmit(EventDetailsEffect.ShowError("Location coordinates are missing"))
 +   }
 }
```
</div>
</details>
</div>
<br>


</div>